### PR TITLE
Remove bootloader logic from `mcu_selection.mk`

### PR DESCRIPTION
**Pull request by zvecr**
## Description

ld filename is now inferred from bootloader type. The find logic now checks for bootloader specific ld before falling back.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

File: keyboards/doio/kb16/rev2/rules.mk [hunk 1]
```diff
@@ -5,14 +5,8 @@ SRC += ./lib/logo.c
...
 # Bootloader selection
-# Cannot use `BOOTLOADER = stm32duino` due to the need to override
-# `MCU_LDSCRIPT`, therefore all parameters need to be specified here manually.
-OPT_DEFS += -DBOOTLOADER_STM32DUINO
-MCU_LDSCRIPT = STM32F103xB_stm32duino_bootloader
-BOARD = STM32_F103_STM32DUINO
-BOOTLOADER_TYPE = stm32duino
-DFU_ARGS = -d 1EAF:0003 -a 2 -R
-DFU_SUFFIX_ARGS = -v 1EAF -p 0003
+BOOTLOADER = stm32duino
```
Comment (fauxpark) [inline keyboards/doio/kb16/rev2/rules.mk:8]: I guess these could be moved to info.json now.

Comment (zvecr) [reply keyboards/doio/kb16/rev2/rules.mk:8]: could do?

File: keyboards/phage_studio/pila87/rules.mk [hunk 2]
```diff
@@ -1,15 +1,8 @@
 # MCU name
 MCU = STM32F103
```
Comment (sigprof) [inline keyboards/phage_studio/pila87/rules.mk:3]: No longer needed here after moving to `info.json`?
```suggestion
```

File: keyboards/rocketboard_16/rules.mk [hunk 3]
```diff
@@ -1,16 +1,8 @@
 # MCU name
 MCU = STM32F103
```
Comment (sigprof) [inline keyboards/rocketboard_16/rules.mk:3]: No longer needed here after moving to `info.json`?
```suggestion
```

**Review by sigprof — CHANGES_REQUESTED** [review]

**Review by sigprof — APPROVED** [review]

**Pull request merged**